src/LanguageServer/Impl/Implementation/Server.cs

```diff
@@ -233,7 +238,36 @@ public sealed partial class Server : IDisposable {
         }
         #endregion
 
-        public void NotifyPackagesChanged(CancellationToken cancellationToken) {
+        public void HandlePathWatchChanges(bool watchSearchPaths) {
+            if (!watchSearchPaths) {
```

Comment:
This logic is sort of confusing. Essentially the value of _watchSearchPaths is always the same as the parameter watchSearchPaths?

Reply:
so you detect when previously we were not watching then we start watching, and when that happens reset the path watcher. Perhaps add a comment to make that more clear, it's hard because the names of the parameter and the field are so similar.

Reply:
I rewrote it to be more obvious.